src/opencl_bitcoin_fmt_plug.c

```diff
@@ -336,43 +364,37 @@ static void *get_salt(char *ciphertext)
 static void set_salt(void *salt)
 {
 	cur_salt = (struct custom_salt *)salt;
+
+	HANDLE_CLERROR(clEnqueueWriteBuffer(queue[gpu_id], mem_salt, CL_FALSE, 0,
+		salt_size, cur_salt, 0, NULL, multi_profilingEvent[0]),
+		"Copy salt to gpu");
+	HANDLE_CLERROR(clFlush(queue[gpu_id]), "clFlush failed in set_salt()");
 }
 
 static int crypt_all(int *pcount, struct db_salt *salt)
 {
 	const int count = *pcount;
-	int index = 0;
 
-	if (any_cracked) {
-		memset(cracked, 0, cracked_size);
-		any_cracked = 0;
-	}
-
-#ifdef _OPENMP
-#pragma omp parallel for
-#endif
-	for (index = 0; index < count; index++) {
-		unsigned char *key_iv = inoutbuffer[index].b;
-		SHA512_CTX sha_ctx;
-		SHA512_Init(&sha_ctx);
-		SHA512_Update(&sha_ctx, saved_key[index], strlen(saved_key[index]));
-		SHA512_Update(&sha_ctx, cur_salt->cry_salt, cur_salt->cry_salt_length);
-		SHA512_Final(key_iv, &sha_ctx);
-
-		unsigned int i;
-		for (i = 0; i < 8; i++)
-			inoutbuffer[index].W[i] = JOHNSWAP64(inoutbuffer[index].W[i]);
-	}
-
-#if 1
 	size_t *lws = local_work_size ? &local_work_size : NULL;
 
 	global_work_size = GET_NEXT_MULTIPLE(count, local_work_size);
 
-	// Copy data to gpu
-	BENCH_CLERROR(clEnqueueWriteBuffer(queue[gpu_id], mem_inout, CL_FALSE, 0,
-		inoutsize, inoutbuffer, 0, NULL, multi_profilingEvent[0]),
-		"Copy data to gpu");
+	in_size = sizeof(password_t) * global_work_size;
+	cracked_size = sizeof(*cracked) * global_work_size;
```

Comment:
Since we have these globals, it felt logical to update them and use them here as well.

Reply:
OK, since we don't need them when freeing.